From d66de0f51b5df812cc31499250a3c629880302ec Mon Sep 17 00:00:00 2001
From: Kati Kankaanpaa <kati.kankaanpaa@digia.com>
Date: Thu, 7 May 2015 13:56:41 -0700
Subject: [PATCH] Fix compile error in XCB when XInput version < 2.2

Add version guards to prevent compiler error:
'XITouchClass' was not declared in this scope
in systems having XInput version < 2.2.

Change-Id: Ib1308f29ef97288eb994ab8bdd668199ca2ee1d7
Reviewed-by: Laszlo Agocs <laszlo.agocs@theqtcompany.com>
---
 src/plugins/platforms/xcb/qxcbconnection_xi2.cpp | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/plugins/platforms/xcb/qxcbconnection_xi2.cpp b/src/plugins/platforms/xcb/qxcbconnection_xi2.cpp
index d1b3ead..2895a27 100644
--- qtbase/src/plugins/platforms/xcb/qxcbconnection_xi2.cpp
+++ qtbase/src/plugins/platforms/xcb/qxcbconnection_xi2.cpp
@@ -175,9 +175,11 @@ void QXcbConnection::xi2SetupDevices()
             case XIKeyClass:
                 qCDebug(lcQpaXInputDevices) << "   it's a keyboard";
                 break;
+#ifdef XCB_USE_XINPUT22
             case XITouchClass:
                 // will be handled in deviceForId()
                 break;
+#endif
             default:
                 qCDebug(lcQpaXInputDevices) << "   has class" << devices[i].classes[c]->type;
                 break;
